### PR TITLE
The Graph: index events

### PR DESCRIPTION
**Pull request by moconnell**

### CODE REVIEW

File: graph/src/hypercert-minter-v-0.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+export function handleBeaconUpgraded(event: BeaconUpgraded): void {}
+
+export function handleImpactClaimed(event: ImpactClaimed): void {
+  let entity = new Hypercert(event.params.id.toString());
+  entity.claimHash = event.params.claimHash;
+  const contributors = [] as string[];
+  for (let i=0; i<event.params.contributors.length; i++)
+  {
+    const address = event.params.contributors[i];
```
Comment (bitbeckers) [inline graph/src/hypercert-minter-v-0.ts:84]: Is this enough for the current needs? We could also make contributors entities and get a many-to-many relationship

File: graph/src/hypercert-minter-v-0.ts [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+export function handleWorkScopeAdded(event: WorkScopeAdded): void {
+  let entity = new WorkScope(event.params.id);
+  entity.text = event.params.text;
+  entity.save();
+}
+
+export function handleTransferBatch(event: TransferBatch): void {}
+
+export function handleTransferSingle(event: TransferSingle): void {
```
Comment (bitbeckers) [inline graph/src/hypercert-minter-v-0.ts:128]: On transfers make sure to skip the zero-address. There are cases where 0x0 doesn't have any balance, for instance when minting

File: graph/package.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  "name": "hypercerts",
+  "license": "UNLICENSED",
+  "scripts": {
+    "codegen": "graph codegen",
+    "build": "graph build",
+    "deploy": "graph deploy --node https://api.thegraph.com/deploy/ protocol/hypercerts",
+    "create-local": "graph create --node http://localhost:8020/ protocol/hypercerts",
+    "remove-local": "graph remove --node http://localhost:8020/ protocol/hypercerts",
+    "deploy-local": "graph deploy --node http://localhost:8020/ --ipfs http://localhost:5001 protocol/hypercerts",
+    "test": "graph test"
```
Comment (Jipperism) [inline graph/package.json:11]: Probably nice to always run `yarn codegen` when running this command

File: graph/tests/.latest.json [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (Jipperism) [inline graph/tests/.latest.json:1]: Should this file be added to `.gitignore`? When running the tests, this got updated automatically

**Review by Jipperism — CHANGES_REQUESTED** [review]

File: graph/src/hypercert-minter-v-0.ts [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+  const toId = `${to}-${tokenId}`;
+  let balanceTo = HypercertBalance.load(toId);
+  if (!balanceTo) {
+    balanceTo = new HypercertBalance(toId);
+    balanceTo.token = tokenId;
+    balanceTo.amount = BigInt.fromI32(0);
+  }
+
+  balanceTo.amount = balanceTo.amount + event.params.value;
```
Comment (Jipperism) [inline graph/src/hypercert-minter-v-0.ts:147]: I think this should be `balanceTo.amount.plus(event.params.value)`. Same for other places where `BigInt`s are being added or substracted.

**Review by Jipperism — CHANGES_REQUESTED** [review]

**Comment by Jipperism** [comment]
Also, it might be nice to add the `graph/build` folder to gitignore? Doesn't seem like that would belong in the repo?

File: graph/tests/.latest.json (hunk 4 above)
Comment (moconnell) [reply graph/tests/.latest.json:1]: added to `.gitignore`

File: graph/package.json (hunk 3 above)
Comment (moconnell) [reply graph/package.json]: added

File: graph/src/hypercert-minter-v-0.ts (hunk 2 above)
Comment (moconnell) [reply graph/src/hypercert-minter-v-0.ts:128]: Have updated

File: graph/src/hypercert-minter-v-0.ts (hunk 5 above)
Comment (moconnell) [reply graph/src/hypercert-minter-v-0.ts]: updated

**Comment by moconnell** [comment]
> Also, it might be nice to add the `graph/build` folder to gitignore? Doesn't seem like that would belong in the repo?

updated

**Pull request merged**